ksql-engine/src/main/java/io/confluent/ksql/planner/plan/AggregateNode.java

```diff
@@ -404,9 +391,15 @@ private ResolveToInternalRewriter() {
         }
 
         final boolean isAggregate = node.getReference().isAggregate();
+        final boolean windowBounds = SchemaUtil.isWindowBound(node.getReference());
```

Comment:
double checking - is there a problem with aggregating on window bounds of the source? e.g. I'm consuming from a windowed stream and performing an aggregation: `SELECT COUNT_DISTINCT(source.WINDOW_START) FROM source`

Reply:
Yes there is a problem. It's not currently possible.  See https://github.com/confluentinc/ksql/issues/4397 for details.